cmake/linker/ld/target_relocation.cmake

```diff
@@ -9,6 +9,14 @@ macro(toolchain_ld_relocation)
   set(MEM_RELOCATION_SRAM_BSS_LD
        "${PROJECT_BINARY_DIR}/include/generated/linker_sram_bss_relocate.ld")
   set(MEM_RELOCATION_CODE "${PROJECT_BINARY_DIR}/code_relocation.c")
+  if(CONFIG_ARM)
+    set(MEM_REGION_DEFAULT_RAM SRAM)
+  elseif(CONFIG_RISCV)
+    set(MEM_REGION_DEFAULT_RAM RAM)
+  else()
```

Comment:
If you want you can submit a new PR for this but can we please unify the nomenclature between RISCV and ARM in the linker scripts instead of having `if-else` around?

Reply:
Opened #49438 for that, which I will mark as ready for review once this is merged (since it builds on top of this branch).